app/src/main/java/app/revanced/integrations/boostforreddit/FixSLinksPatch.java

```diff
@@ -0,0 +1,93 @@
+package app.revanced.integrations.boostforreddit;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.os.StrictMode;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import app.revanced.integrations.shared.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+
+public final class FixSLinksPatch {
+    public static String getUserAccessToken(Context context) {
+        File dbFile = context.getDatabasePath("reddit.db");
+        if (!dbFile.exists()) {
+            Logger.printInfo(() -> "Reddit database is not present at " + dbFile.getPath());
+            return null;
+        }
+        SQLiteDatabase redditDb = SQLiteDatabase.openDatabase(
+                dbFile.getPath(),
+                null, SQLiteDatabase.OPEN_READONLY);
+        Cursor tokenCursor = redditDb.rawQuery("SELECT AccessToken FROM User", null);
+        if (tokenCursor.getCount() == 0) {
+            Logger.printInfo(() -> "No authorized users found");
+            return null;
+        }
+        tokenCursor.moveToFirst();
+        String token = tokenCursor.getString(0);
+        tokenCursor.close();
+        redditDb.close();
+        return token;
+    }
+
+    public static String resolveSLink(Context context, String link) {
+        if (link.matches(".*reddit\\.com/r/[^/]+/s/[^/]+")) {
+            Logger.printInfo(() -> "Resolving " + link);
+            try {
+                HttpURLConnection connection = getHttpURLConnection(context, link);
+
+                // Disable strict mode in order to allow network access on the main thread.
+                // This is not ideal, but it's the easiest solution for now.
+                final var currentPolicy = StrictMode.getThreadPolicy();
+                StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+                StrictMode.setThreadPolicy(policy);
+
+                connection.connect();
+                String location = connection.getHeaderField("location");
+                connection.disconnect();
+
+                // Restore the original strict mode policy.
+                StrictMode.setThreadPolicy(currentPolicy);
+
+                Logger.printInfo(() -> "Resolved " + link + " -> " + location);
+                // For some reason using requireNotNull or similar ends up in java.lang.ExceptionInInitializerError,
```

Comment:
Probably a throwable is raised instead of an exception

Reply:
Error still gets logged by exception handler however